### PR TITLE
feat: add support for MCLI_PROFILE

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

Add support for the MCLI_PROFILE env variable
## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/root/builder.go [hunk 1]
```diff
@@ -65,7 +67,17 @@ no additional shell configuration is necessary, see https://docs.brew.sh/Shell-C
 // This is important in particular for Atlas as it dynamically sets flags for cluster creation and
 // this can be slow to timeout on environments with limited internet access (Ops Manager)
 func Builder(profile *string, argsWithoutProg []string) *cobra.Command {
-	rootCmd = cli.Builder()
+	rootCmd := &cobra.Command{
```
Comment (gssbzn) [inline internal/cli/root/builder.go:70]: moving this in from the `cli.go` file they were both serving the same kind of functionality

**Review by savannaendicott — APPROVED** [review]
LGTM!

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Comment by gssbzn** [comment]
failing test transient from shared testing env

**Pull request merged**